### ISSUE TITLE
Can we use a Cantaloupe iiif endpoint to load images in non iiif applications?

### ISSUE THREAD

**Issue by imoutsatsos** [medusa-project/cantaloupe#243]
Hi Alex; Long time happily working with Cantaloupe v 3.3. 

It has served extremely well most of my requirements within the iiif framework. I'm now trying something different that I had hoped it was straight forward but perhaps only because my basic assumptions were wrong.

I'm hoping to use Cantaloupe as the image source for our main scientific application for image processing, CellProfiler http://cellprofiler.org/

The application (written in python and using the urllib2.py module) loads the 16 bit TIF files and applies various image processing algorithms to extract image feature measurements. It can load image files from a URL, or local file path. 

I was hoping I could replace the URL with a Cantaloupe iiif endpoint URI to load the images. Using this configuration the application hangs and eventually reports a '504 Gateway Timeout' error. 

![image](https://user-images.githubusercontent.com/3408226/46968701-ebc95900-d081-11e8-8cfd-25cd459c915a.png)

I also get exactly the same error trying to get the tif file from the Cantaloupe iiif end point using 'curl'
`curl http://myserver.net:8182/iiif/2/%2F%2FPC%2FYOKOGAWA%2FZh%2FNpack%2FZ20180411_162332%2FCM00620249%2FA02Z01C02.tif/full/full/0/default.jpg -D header_iiif.txt>default.jpg`

Nonetheless, the Cantaloupe image URI (default.jpg or default.tif ) works perfectly in various browsers (Chrome, IE, MS-Edge). When the 'jpg' file type is used it displays the image in the browser, when 'tif' is used the browser downloads the tif image from Cantaloupe.

So, I am now questioning my simple assumption that iiif endpoints can be used as a simple URL to an image file. Do you think that the endpoints are too complex to be treated as such, and so only a iiif compatible application understands how to use them correctly. Is anything else like `urllib.py` and `curl` getting hopelessly confused?

Any feedback on this question would be greatly appreciated.
Thanks and best regards
Ioannis

**Comment by adolski** [comment]
Hi @imoutsatsos,

I'm not sure what's happening here, but I don't think your assumptions are wrong. There is nothing special about image endpoints, and any HTTP client should be able to access them. I can't reproduce this with curl 7.54.0 + Cantaloupe 3.3.5.

**Comment by imoutsatsos** [comment]
Hi again @adolski 
I also posted a similar question on the iiif slack channel. There was some suspicion that the %escaped slashes in the image identifier were causing proxy server issues. Cantaloupe of course provides alternatives for this but as it turns out it was simply a combination of finding the right proxy server to use and the right protocol.  Once I figures this out, the iiif Cantaloupe endpoints worked fine to retrieve images. Here is a gist I wrote to self. https://gist.github.com/imoutsatsos/b95ae233885985d5acf29c55d25c44b4 
Closing issue. Thanks! (by the way i have now also upgraded to Cantaloupe v4.0.1)
